blacklight-cornell/app/helpers/search_history_helper.rb

```diff
@@ -177,18 +178,25 @@ def build_search_history_url(params, search_type)
     end
 
     # Date range facets
-    if dr_row.present? && dr_row[:pub_date_facet].is_a?(Hash)
-      begin_val = dr_row[:pub_date_facet][:begin]
-      end_val   = dr_row[:pub_date_facet][:end]
-      if begin_val.present? || end_val.present?
-        query[:range] ||= {}
-        query[:range][:pub_date_facet] = {}
-        query[:range][:pub_date_facet][:begin] = begin_val if begin_val.present?
-        query[:range][:pub_date_facet][:end]   = end_val   if end_val.present?
-      end
+    pub_date_facet = dr_row[:pub_date_facet]         || {}
+    begin_val      = pub_date_facet[:begin].presence || ""
+    end_val        = pub_date_facet[:end].presence   || ""
+
+    if search_type == :advanced
+      query[:range] = { pub_date_facet: { begin: begin_val, end: end_val } }
+    else
+      query[:range] = { pub_date_facet: { begin: begin_val, end: end_val } } if begin_val.present? && end_val.present?
     end
 
-    "#{base_path}?#{query.to_query}" # Search URL with query params
+    # Build Query String -------------------------------------------------------
```

Comment:
Updated to properly set the parameters in the query in the right order. This prevents duplicate search history results when you click on the link